### PR TITLE
Selector reports

### PR DESCRIPTION
**Pull request by mmuzikar**
//skip-ci

### CODE REVIEW

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/hooks/
  SelectorSnooper.java:38:31: Name 'screenShotFolder' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageReporter.java:20:57: Name 'reports' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - Database | Smoke - Periodic invocation to Insert 
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageInfo.java:30:46: 'url' hides a field. [HiddenField]
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageInfo.java:30:50: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageReporter.java:38:97: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageReporter.java:59:39: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageReporter.java:67:43: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageReporter.java:70:65: '{' is not preceded with whitespace. [WhitespaceAround]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageReporter.java:86:57: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/reports/selectors/
  SelectorUsageReporter.java:88:29: '{' is not preceded with whitespace. [WhitespaceAround]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/hooks/
  SelectorSnooper.java:41:48: Name 'reporter' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Validate all credentials 
Connection - CRUD | Create and delete 
Connection - CRUD | Kebab menu 
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Validate all credentials 
Customization - API Connector - ApicurIO GUI | Check if operations change is propagated into connector review page from ApicurIO GUI 
Customization - API Connector - ApicurIO GUI | Check if operations change with an error is propagated into connector review page from ApicurIO GUI 
Customization - API Connector - ApicurIO GUI | Check if operations change is correctly not propagated into connector when changes from ApicurIO GUI are not saved 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - no security 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - basic security 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - API key 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - OAuth 2 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | Conditional flows used in API Provider 
Concur Connector | Check message of concur get list action 
Concur Connector | Check message of concur get listitems action 
Concur Connector | Check if create and delete listitems actions work together 
Conditional flows - content base routing | Conditional flows icon test - move up 
Conditional flows - content base routing | Conditional flows icon test - delete 
Conditional flows - content base routing | Conditional flows icon test multiple moves 
Conditional flows - content base routing | Conditional flows - functional test 
Conditional flows - content base routing | Conditional flows - delete step 
Conditional flows - content base routing | Conditional flows - edit and update step 
Conditional flows - content base routing | Conditional flows - multiple flows 
Conditional flows - content base routing | Conditional flows - dropdown test 
Integration - Databucket | Create extended 
Integration - data shapes propagation | Data shapes propagation into conditional flow 
Integration - DB to DB mysql | Read and create operations 
Integration - Dropbox to Dropbox | Create 
Google Sheets Connector | Append messages from DB 
Google Sheets Connector | Update messages from DB - Columns 
Google Sheets Connector | Copy big spreadsheet 
Google Sheets Connector | Copy Big SS to DB and back using split/aggregate 
Integration - Import Export | Import and export drag and drop 
Integration - Import Export | Import from different syndesis instance 
Integration - DB to DB | Check starting integration status on detail page 
Integration - DB to DB | Check starting integration status on integration list 
Integration - DB to DB | Check starting integration status on home page 
Kudu connector | Insert data into kudu table 
Kudu connector | Scan kudu table 
Kudu connector | Kudu integration with mqtt 
Log Connector | Check log message with body: true and context: true 
Log Connector | Check log message with body: true and context: false 
Log Connector | Check log message with body: false and context: true 
Log Connector | Check log message with body: false and context: false 
Log Connector | Check log without message 
Integration - MQTT to MQTT | Publish subscribe on topic 
OData Connector | Read entity collection from OData service tests 
OData Connector | Read single entity from OData service tests 
OData Connector | Read property of entity from OData service tests 
OData Connector | Read with $expand query from OData service tests 
OData Connector | Create entity in OData service 
OData Connector | Delete entity in OData service 
OData Connector | Read wrong_datashape from OData service tests 
OData Connector | Read calling_null from OData service tests 
OData Connector | Read NPE from OData service tests 
OData Connector | Read string predicate without quotes from OData service tests 
OData Connector | Read enum mapping from OData service tests 
OData Connector | OData connector needs server certificate 
Integration - Openwire to Openwire | Publish subscribe on queue 
Integration - Salesforce to DB | Create 
Integration - ServiceNow-amq/log | Get incident from service now 
Integration - ServiceNow-amq/log | Create incident from syndesis integration 
Slack Connector | Check that slack received a message from an integration 
Slack Connector | Check that slack message is saved into DB 
Slack Connector | Check Maximum Messages to Retrieve and Delay function in SLACK consumer 
Integration - DB to DB with split/aggregate | Split and aggregate integration flow 
SQL Connector | Check SQL pulls all rows 
SQL Connector | Check SQL connector with predicate id < :#id 
SQL Connector | Check SQL connector with predicate id > :#id 
SQL Connector | Check SQL connector with predicate id <= :#id 
SQL Connector | Check SQL connector with predicate id >= :#id 
SQL Connector | Check SQL connector with predicate id != :#id 
SQL Connector | Check SQL connector with predicate id <= :#id AND id > 1 
SQL Connector | Check SQL connector with predicate id BETWEEN :#id AND 4 
SQL Connector | Check SQL connector with predicate id <= :#id OR id = 5 
SQL Connector | Check SQL connector with predicate :#id > id 
SQL Connector | Check SQL connector with predicate (id+1) < :#id 
SQL Connector | Check SQL connector with predicate id in (1, :#id) 
SQL Connector | Check SQL connector with predicate id = floor(:#id) 
SQL Connector | Return generated keys for INSERT statement 
SQL Connector | Return generated keys for INSERT statement from start step 
Templates | Send an Email with text formatted by Mustache template 
Templates | Send an Email with text formatted by Freemarker template 
Templates | Send an Email with text formatted by Velocity template 
Templates | Send an Email with text formatted by Mustache template uploaded from resources/templates/mustache.txt 
Templates | Verify that the code editor for templates is working correctly 
Integration - Twitter Direct Messages | Receive direct message on twitter 
Integration - Twitter Direct Messages | Send direct message on twitter 
Integration - Twitter to Database | Create integration from twitter to database 
Integration - Twitter to Salesforce | Create 
Integration - Webhook to API connector | Webhook to Custom API connector 
Integration - Webhook to API connector | Test to reproduce #3727 
Integration - Webhook to DB | Webhook to db with middle db step 
3scale integration | 3scale discovery annotations 
Activity | Check activity 
Activity | Check error 
Activity | Check whether shows error when data mapper is before 
Metrics | Check metrics 
Metrics | Check error 
Login logout | Log out 
Login logout | Log out and log in 
Test tagging integration in CI CD dialog | Test creating new tag 
Test tagging integration in CI CD dialog | Test updating tag in CI CD dialog 
Test tagging integration in CI CD dialog | Test checking tag in CI CD dialog 
Test tagging integration in CI CD dialog | Test unchecking tag in CI CD dialog and whether it is still exist when no integration uses it 
Test tagging integration in CI CD dialog | Test deleting tag in CI CD dialog 
Test tagging integration in CI CD dialog | Test check and delete multiple tag across integrations 
Test tagging integration in CI CD dialog | Test check number of tag 
Import and export integration test | Test export the integrations according to tag and import it again 
Import and export integration test | Test export the all integrations and import it again 
Settings OAuth | Fill OAuth settings and check immediately 
Settings OAuth | Fill OAuth settings and check after changing page 
Settings OAuth | Fill OAuth settings, remove and check immediately 
Settings OAuth | Fill OAuth settings, remove and check after changing page 
Settings OAuth | Fill Gmail OAuth settings, create connection and validate 
Settings OAuth | Fill Google Calendar OAuth settings, create connection and validate 
Settings OAuth | Fill Twitter OAuth settings, create connection and validate 
Support page | Version 
Support page | Export diagnostic of all 
Support page | Export diagnostic of single integration 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Customization - API Connector - ApicurIO GUI | Check if operations change is propagated into connector review page from ApicurIO GUI 
Customization - API Connector - ApicurIO GUI | Check if operations change with an error is propagated into connector review page from ApicurIO GUI 
Customization - API Connector - ApicurIO GUI | Check if operations change is correctly not propagated into connector when changes from ApicurIO GUI are not saved 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - no security 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - basic security 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - API key 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - OAuth 2 
Webhook extension | Check message 
Webhook extension | Check whether UI contains webhook URL 
API Provider Integration | API Provider GET emptycollection 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | Conditional flows used in API Provider 
Concur Connector | Check message of concur get list action 
Concur Connector | Check message of concur get listitems action 
Concur Connector | Check if create and delete listitems actions work together 
Conditional flows - content base routing | Conditional flows icon test - move up 
Conditional flows - content base routing | Conditional flows icon test - delete 
Conditional flows - content base routing | Conditional flows icon test multiple moves 
Conditional flows - content base routing | Conditional flows - functional test 
Conditional flows - content base routing | Conditional flows - delete step 
Conditional flows - content base routing | Conditional flows - edit and update step 
Conditional flows - content base routing | Conditional flows - multiple flows 
Conditional flows - content base routing | Conditional flows - dropdown test 
Integration - data shapes propagation | Data shapes propagation into conditional flow 
Integration - DB to DB mysql | Read and create operations 
Integration - Dropbox to Dropbox | Create 
Google Calendar Connector | Test get events action - poll events updated during runtime 
Google Sheets Connector | Append messages from DB 
Google Sheets Connector | Update messages from DB - Columns 
Integration - Import Export | Import and export drag and drop 
Integration - Import Export | Import from different syndesis instance 
Integration - DB to DB | Check starting integration status on detail page 
Integration - DB to DB | Check starting integration status on integration list 
Integration - DB to DB | Check starting integration status on home page 
Kudu connector | Insert data into kudu table 
Kudu connector | Scan kudu table 
Kudu connector | Kudu integration with mqtt 
Log Connector | Check log message with body: true and context: true 
Log Connector | Check log message with body: true and context: false 
Log Connector | Check log message with body: false and context: true 
Log Connector | Check log message with body: false and context: false 
Log Connector | Check log without message 
Integration - MQTT to MQTT | Publish subscribe on topic 
OData Connector | Read entity collection from OData service tests 
OData Connector | Read single entity from OData service tests 
OData Connector | Read property of entity from OData service tests 
OData Connector | Read with $expand query from OData service tests 
OData Connector | Create entity in OData service 
OData Connector | Delete entity in OData service 
OData Connector | Read wrong_datashape from OData service tests 
OData Connector | Read calling_null from OData service tests 
OData Connector | Read NPE from OData service tests 
OData Connector | Read string predicate without quotes from OData service tests 
OData Connector | Read enum mapping from OData service tests 
OData Connector | OData connector needs server certificate 
Integration - Openwire to Openwire | Publish subscribe on queue 
Integration - Salesforce to DB | Create 
Integration - ServiceNow-amq/log | Get incident from service now 
Integration - ServiceNow-amq/log | Create incident from syndesis integration 
Slack Connector | Check that slack received a message from an integration 
Slack Connector | Check that slack message is saved into DB 
Slack Connector | Check Maximum Messages to Retrieve and Delay function in SLACK consumer 
Integration - DB to DB with split/aggregate | Split and aggregate integration flow 
SQL Connector | Check SQL pulls all rows 
SQL Connector | Check SQL connector with predicate id < :#id 
SQL Connector | Check SQL connector with predicate id > :#id 
SQL Connector | Check SQL connector with predicate id <= :#id 
SQL Connector | Check SQL connector with predicate id >= :#id 
SQL Connector | Check SQL connector with predicate id != :#id 
SQL Connector | Check SQL connector with predicate id <= :#id AND id > 1 
SQL Connector | Check SQL connector with predicate id BETWEEN :#id AND 4 
SQL Connector | Check SQL connector with predicate id <= :#id OR id = 5 
SQL Connector | Check SQL connector with predicate :#id > id 
SQL Connector | Check SQL connector with predicate (id+1) < :#id 
SQL Connector | Check SQL connector with predicate id in (1, :#id) 
SQL Connector | Check SQL connector with predicate id = floor(:#id) 
SQL Connector | Return generated keys for INSERT statement 
SQL Connector | Return generated keys for INSERT statement from start step 
Templates | Send an Email with text formatted by Mustache template 
Templates | Send an Email with text formatted by Freemarker template 
Templates | Send an Email with text formatted by Velocity template 
Templates | Send an Email with text formatted by Mustache template uploaded from resources/templates/mustache.txt 
Templates | Verify that the code editor for templates is working correctly 
Integration - Twitter Direct Messages | Receive direct message on twitter 
Integration - Twitter Direct Messages | Send direct message on twitter 
Integration - Twitter to Database | Create integration from twitter to database 
Integration - Twitter to Salesforce | Create 
Integration - Webhook to API connector | Webhook to Custom API connector 
Integration - Webhook to API connector | Test to reproduce #3727 
Integration - Webhook to DB | Webhook to db with middle db step 
3scale integration | 3scale discovery annotations 
Activity | Check activity 
Activity | Check error 
Activity | Check whether shows error when data mapper is before 
Metrics | Check metrics 
Metrics | Check error 
Login logout | Log out 
Login logout | Log out and log in 
Test tagging integration in CI CD dialog | Test creating new tag 
Test tagging integration in CI CD dialog | Test updating tag in CI CD dialog 
Test tagging integration in CI CD dialog | Test checking tag in CI CD dialog 
Test tagging integration in CI CD dialog | Test unchecking tag in CI CD dialog and whether it is still exist when no integration uses it 
Test tagging integration in CI CD dialog | Test deleting tag in CI CD dialog 
Test tagging integration in CI CD dialog | Test check and delete multiple tag across integrations 
Test tagging integration in CI CD dialog | Test check number of tag 
Import and export integration test | Test export the integrations according to tag and import it again 
Import and export integration test | Test export the all integrations and import it again 
Settings OAuth | Fill OAuth settings and check immediately 
Settings OAuth | Fill OAuth settings and check after changing page 
Settings OAuth | Fill OAuth settings, remove and check immediately 
Settings OAuth | Fill OAuth settings, remove and check after changing page 
Settings OAuth | Fill Gmail OAuth settings, create connection and validate 
Settings OAuth | Fill Google Calendar OAuth settings, create connection and validate 
Settings OAuth | Fill Twitter OAuth settings, create connection and validate 
Support page | Version 
Support page | Export diagnostic of all 
Support page | Export diagnostic of single integration 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Customization - API Connector - ApicurIO GUI | Check if operations change is propagated into connector review page from ApicurIO GUI 
Customization - API Connector - ApicurIO GUI | Check if operations change with an error is propagated into connector review page from ApicurIO GUI 
Customization - API Connector - ApicurIO GUI | Check if operations change is correctly not propagated into connector when changes from ApicurIO GUI are not saved 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - no security 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - basic security 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - API key 
Customization - API Connector - ApicurIO GUI | Check apicurio security settings - OAuth 2 
Integration - AMQ to REST | Publish subscribe on topic 
Integration - DB to API | Create an integration with custom API connector as finish step 
Integration - DB to API | Create an integration with custom API connector as middle step 
API Provider Integration | API Provider POST existing  - known issue
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | Conditional flows used in API Provider 
Concur Connector | Check message of concur get list action 
Concur Connector | Check message of concur get listitems action 
Concur Connector | Check if create and delete listitems actions work together 
Conditional flows - content base routing | Conditional flows icon test - move up 
Conditional flows - content base routing | Conditional flows icon test - delete 
Conditional flows - content base routing | Conditional flows icon test multiple moves 
Conditional flows - content base routing | Conditional flows - functional test 
Conditional flows - content base routing | Conditional flows - delete step 
Conditional flows - content base routing | Conditional flows - edit and update step 
Conditional flows - content base routing | Conditional flows - multiple flows 
Conditional flows - content base routing | Conditional flows - dropdown test 
Integration - Databucket | Create 
Integration - Databucket | Create extended 
Integration - data shapes propagation | Data shapes propagation into conditional flow 
Integration - DB to DB mysql | Read and create operations 
Integration - DB to DB with extension | Create 
Integration - DB to DB | Sql query checker 
Integration - DB to DB | Inserting multiple rows 
Google Sheets Connector | Append messages from DB 
Google Sheets Connector | Update messages from DB - Columns 
Google Sheets Connector | Copy Big SS to DB and back using split/aggregate 
Integration - Import Export | Import and export drag and drop 
Kudu connector | Insert data into kudu table 
OData Connector | Create entity in OData service 
OData Connector | Read calling_null from OData service tests 
Integration - ServiceNow-amq/log | Get incident from service now 
Integration - ServiceNow-amq/log | Create incident from syndesis integration 
Slack Connector | Check that slack received a message from an integration 
Slack Connector | Check that slack message is saved into DB 
Slack Connector | Check Maximum Messages to Retrieve and Delay function in SLACK consumer 
SQL Connector | Check SQL connector with predicate id < :#id 
SQL Connector | Check SQL connector with predicate id > :#id 
SQL Connector | Check SQL connector with predicate id <= :#id 
SQL Connector | Check SQL connector with predicate id >= :#id 
SQL Connector | Check SQL connector with predicate id != :#id 
SQL Connector | Check SQL connector with predicate id <= :#id AND id > 1 
SQL Connector | Check SQL connector with predicate id BETWEEN :#id AND 4 
SQL Connector | Check SQL connector with predicate id <= :#id OR id = 5 
SQL Connector | Check SQL connector with predicate :#id > id 
SQL Connector | Check SQL connector with predicate (id+1) < :#id 
SQL Connector | Check SQL connector with predicate id in (1, :#id) 
SQL Connector | Check SQL connector with predicate id = floor(:#id) 
Templates | Verify that the code editor for templates is working correctly 
Integration - Twitter Direct Messages | Send direct message on twitter 
Integration - Twitter to Salesforce | Create 
3scale integration | 3scale discovery annotations 
Test tagging integration in CI CD dialog | Test creating new tag 
Test tagging integration in CI CD dialog | Test updating tag in CI CD dialog 
Test tagging integration in CI CD dialog | Test unchecking tag in CI CD dialog and whether it is still exist when no integration uses it 
Import and export integration test | Test export the integrations according to tag and import it again 
Import and export integration test | Test export the all integrations and import it again 
Support page | Version 
Support page | Export diagnostic of single integration 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Validate all credentials 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: ui-tests/src/test/java/io/syndesis/qe/hooks/SelectorSnooper.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            }
+        }
+    }
+
+    private static boolean usesForbiddenCssCharacters(String id) {
+        Matcher matcher = FORBIDDEN_CSS_CHARACTERS.matcher(id);
+        return matcher.matches();
+    }
+
+    public static final class WebElementSelectorDetector extends WebElementSelector {
```
Comment (asmigala) [inline ui-tests/src/test/java/io/syndesis/qe/hooks/SelectorSnooper.java]: This class could be made private I believe

File: ui-tests/src/test/java/io/syndesis/qe/utils/ExcludeFromSelectorReports.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package io.syndesis.qe.utils;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
```
Comment (asmigala) [inline ui-tests/src/test/java/io/syndesis/qe/utils/ExcludeFromSelectorReports.java]: Maybe we could consider targeting whole classes as well, that would simplify the apicurito and atlasmap steps above. Not necessary though

File: ui-tests/src/test/resources/templates/data_test_id_template.vm [hunk 3]
```diff
@@ -0,0 +1,91 @@
+<html>
```
Comment (asmigala) [inline ui-tests/src/test/resources/templates/data_test_id_template.vm]: Thumbs up on using a templating engine :D

File: ui-tests/src/test/resources/templates/url_report_template.vm [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                        <b>URL: $url</b> 
+                        </td>
+                    </tr>
+                    #foreach( $report in $reports.get($url) )
+                        <tr>
+                            <td class="result-body result-selector">
+                                $report.selector
+                            </td>
+                            <td class="result-body">
+                                #set ( $ghUrl = $utils.getGithubURL($report.stackTrace))
```
Comment (asmigala) [inline ui-tests/src/test/resources/templates/url_report_template.vm]: awesome

File: utilities/src/main/java/io/syndesis/qe/TestConfiguration.java [hunk 5]
```diff
@@ -219,6 +221,10 @@ public static int stateCheckInterval() {
         return Integer.parseInt(get().readValue(STATE_CHECK_INTERVAL));
     }
 
+    public static boolean snoopselectors() {
```
Comment (asmigala) [inline utilities/src/main/java/io/syndesis/qe/TestConfiguration.java]:
```suggestion
    public static boolean snoopSelectors() {
```

maybe?

File: ui-tests/src/test/java/io/syndesis/qe/hooks/SelectorSnooper.java [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        wait.until(ExpectedConditions.attributeContains(el, "style", "2px solid red"));
+        File tmpImage = Screenshots.takeScreenShotAsFile();
+        String fileName = Instant.now().toString() + ".png";
+        String path = SCREEN_SHOT_FOLDER.resolve(fileName).toString();
+        try {
+            FileUtils.copyFile(tmpImage, FileUtils.getFile(path));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        //There is no way to reliably wait for change of style, so alert it is
```
[the remark this answers is not in the document]
Comment (mmuzikar) [reply ui-tests/src/test/java/io/syndesis/qe/hooks/SelectorSnooper.java]: Yes I did :sunglasses: it's easy to check if the style is present, but checking for inline style not being present/not containing something didn't work well. The alert is way more reliable timing-wise. I will change the comment though.

File: ui-tests/src/test/java/io/syndesis/qe/utils/ExcludeFromSelectorReports.java (hunk 2 above)
Comment (mmuzikar) [reply ui-tests/src/test/java/io/syndesis/qe/utils/ExcludeFromSelectorReports.java]: I also thought about that, but apicurito steps have some syndesis related methods... and also if some methods get added in the future there's no clear indication that some bytecode manipulation is performed per method... and also it'd be more work :D

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**